lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -904,7 +904,7 @@ SILCombiner::buildConcreteOpenedExistentialInfoFromSoleConformingType(
         (archetypeTy->getConformsTo().size() == 1)) {
       PD = archetypeTy->getConformsTo()[0];
     } else if (ArgType.isExistentialType() && !ArgType.isAnyObject() &&
-               !SwiftArgType->isAny()) {
+               !SwiftArgType->isAny() && SwiftArgType->getAnyNominal()) {
       PD = dyn_cast<ProtocolDecl>(SwiftArgType->getAnyNominal());
```

Comment:
This is equivalent to `dyn_cast_or_null` on the next line.  But it sounds like this code needs to handle protocol compositions in a more principled way.  It seems like there’s a missing case here.

Reply:
What missing case is that?

Reply:
Not sure @atrick is more familiar with this code.